modules/discretedata/dataset.h

```diff
@@ -0,0 +1,209 @@
+/*********************************************************************************
+*
+* Inviwo - Interactive Visualization Workshop
+*
+* Copyright (c) 2012-2018 Inviwo Foundation
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*********************************************************************************/
+
+#pragma once
+
+#include <discretedata/discretedatamoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/datastructures/datatraits.h>
+
+#include "channels/bufferchannel.h"
+#include "connectivity/connectivity.h"
+
+namespace inviwo {
+namespace dd {
+
+typedef std::shared_ptr<Channel> SharedChannel;
```

Comment:
From my experience, i think it's better not to typedef the shared_ptr like this, the user really needs to know that it is a std::shared_ptr to be able to create is using for example std::make_shared. And also for how to think about it with respect to lifetime etc. so I would prefer to use std::shared_ptr<Channel> std::shared_ptr<const Channel> in the interface instead of the typedefs.